### PR TITLE
Forgotten plugins

### PR DESCRIPTION
**Pull request by Florent2305**
- Optimus (from Asclepios)
 - DWIBrainMask ( accidentally present but removed from cmake)

### CODE REVIEW

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+public:
+    QSpinBox * iterationsBox;
+    QSpinBox * binsBox;
+    QSpinBox * samplesBox;
+    QSpinBox * interpolationsBox;
+    QDoubleSpinBox * rhoStartBox;
+    QDoubleSpinBox * rhoEndBox;
+    QDoubleSpinBox * scalingCoeffBox;
+    QComboBox * interpolatorTypeBox;
+    medProgressionStack * progression_stack;
```
Comment (mathildemerle) [inline src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:44]: the actual toolboxes use a common progression stack, you can remove this line

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    d->progression_stack = new medProgressionStack(widget);
+
+    this->addWidget(widget);
+    this->addWidget(runButton);
+    this->addWidget(d->progression_stack);
+
+    //enable about plugin. Constructor called after the plugin has been registered, go ahead call it.
+    medPluginManager* pm = medPluginManager::instance();
+    dtkPlugin* plugin = pm->plugin("itkProcessRegistrationOptimusPlugin");
+    setAboutPluginButton(plugin);
```
Comment (mathildemerle) [inline src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:144]: these setAboutblabla methods can be removed

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+    // progression stack
+    d->progression_stack = new medProgressionStack(widget);
+
+    this->addWidget(widget);
+    this->addWidget(runButton);
+    this->addWidget(d->progression_stack);
+
+    //enable about plugin. Constructor called after the plugin has been registered, go ahead call it.
+    medPluginManager* pm = medPluginManager::instance();
```
Comment (mathildemerle) [inline src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:142]: these 2 lines seem wrong. This is already called in `itkProcessRegistrationOptimusToolBox::plugin()` with an other name (Optimus), and these lines are not used here in other toolboxes

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        connect(runProcess, SIGNAL(success(QObject*)), this, SIGNAL(success()));
+        connect(runProcess, SIGNAL(failure(QObject*)), this, SIGNAL(failure()));
+        connect(runProcess, SIGNAL(cancelled(QObject*)), this, SIGNAL(failure()));
+        //First have the moving progress bar,
+        //and then display the remaining % when known
+        connect(runProcess, SIGNAL(activate(QObject*, bool)),
+            d->progression_stack, SLOT(setActive(QObject*, bool)));
+
+        medJobManagerL::instance()->registerJobItem(runProcess, d->process->identifier());
+        QThreadPool::globalInstance()->start(dynamic_cast<QRunnable*>(runProcess));
```
Comment (mathildemerle) [inline src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:254]: replace these paragraphs by `this->addConnectionsAndStartJob(runProcess);` and if needed `enableOnProcessSuccessImportOutput(runProcess, false);`

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        // process->setMyWonderfullParameter(fronTheGui);
+
+        // ----
+
+        d->process->setFixedInput(fixedData);
+        d->process->setMovingInput(movingData);
+
+        medRunnableProcess *runProcess = new medRunnableProcess;
+        runProcess->setProcess(d->process);
+
```
Comment (mathildemerle) [inline src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:241]: add `this->setToolBoxOnWaitStatus();`

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp (hunk 1 above)
Comment (Florent2305) [reply src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp]: ok

File: src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp (hunk 5 above)
Comment (Florent2305) [reply src/plugins/legacy/itkProcessRegistrationOptimus/itkProcessRegistrationOptimusToolBox.cpp:231]: done

**Review by Florent2305 — COMMENTED** [review]
normally I apply all requested changes

**Review by mathildemerle — APPROVED** [review]

**Pull request merged**